### DISCUSSION TITLE
healthcheck.sh has the ROCKET_PORT hardcoded to 80

### DISCUSSION THREAD

**Discussion by CookieUniverse** [dani-garcia/vaultwarden#4733] [Q&A] [answered]
My issue is what's in the title, because of that my container keeps saying it's unhealthy, since it's trying to access port 80 but I use another ROCKET_PORT defined in the environment. Is this a bug? Is there any way I can change it manually?

**Comment by BlackDex** [comment, marked as answer]
If you would read the comments in that script, it isn't hardcoded
https://github.com/dani-garcia/vaultwarden/blob/a4ab014ade53e4e60bda0b9cbce3af9de7eac753/docker/healthcheck.sh#L3-L7

It uses the environment variable if it exists, else it will use 80.
So nothing is hardcoded, it's a fallback.

You are probably running into #4497 .

**Reply by CookieUniverse** [reply]
I'm sorry for missing the comment I was so tunnel visioned that I didn't even notice, but well it wasn't related to #4497 since the it was opening rocket in the right port and websockets were working, it was just the script that wasn't working. But when I checked the compose file I found out that instead of having `ROCKET_PORT=8000`, I had `ROCKET_PORT= 8000` and changing it fixed the problem, somehow the extra space didn't break the websockets but broke the script. Thank you for your answer

**Comment by ebardes** [comment]
I agree that the := idiom only assignes default values if the variable isn't already defined.
However, some shells don't support that idiom. I don't have a list of which do, but can you be sure that "sh" is always "bash" in Docker?

**Reply by CookieUniverse** [reply]
Thank you for your answer, but I already fixed the problem and it was really stupid on my part. The fix is in the reply to @BlackDex

**Reply by BlackDex** [reply]
Well the shell's used in our docker builds do support it 

Even shellcheck says it's fine, and it will complain about POSIX invalid syntax if you set the shebang to be sh.

It might be that BusyBox doesn't, but i have not tried that yet.

**Reply by BlackDex** [reply]
I just checked BusyBox, and that works just fine too. So nothing to worry about.
